internal/controller/bucket/delete_test.go

```diff
@@ -650,3 +757,23 @@ func TestDelete(t *testing.T) {
 		})
 	}
 }
+
+// Unlike NoSuchBucket error or others, aws-sdk-go-v2 doesn't have a specific struct definition for BucketNotEmpty error.
+// So we should define ourselves for testing.
+type bucketNotEmptyError struct{}
+
+func (e bucketNotEmptyError) Error() string {
+	return "BucketNotEmpty: some error"
+}
+
+func (e bucketNotEmptyError) ErrorCode() string {
+	return "BucketNotEmpty"
+}
+
+func (e bucketNotEmptyError) ErrorMessage() string {
+	return "some error"
+}
+
+func (e bucketNotEmptyError) ErrorFault() smithy.ErrorFault {
+	return smithy.FaultUnknown
+}
```

Comment:
Can we keep this in `bucket_helpers.go` and import as needed to avoid duplication?

Reply:
Addressed in https://github.com/linode/provider-ceph/pull/178/commits/73d6faeb1ea6f9ac309285401b9106cc8d0f3ff4